docs/advanced-topics/networkobject-parenting.md

```diff
@@ -0,0 +1,173 @@
+    - Write `m_IsReparented` and `m_LatestParent` fields into a `NetworkBuffer` and send that over to all connected clients with `PARENT_SYNC` message type on `MLAPI_INTERNAL` channel
+
+Transform parent synchronization relies on the initial formation of transforms in the scene hierarchy being identical on all standalone instances.
+
+## NetworkObject Reparenting Rules
+
+A few basic `NetworkObject` reparenting rules are listed below.
+
+:::warning Limiting Non-Networked NetworkObject Transform Parenting
+Rules outlined below are applied and enforced even while not networking (not hosting or connected). Specifically, if you were to try reparenting a `NetworkObject` under a non-`NetworkObject`, that'd be invalid and reverted even though you are not hosting or connected to a server.
+:::
+
+
+### Only A Server (or A Host) Can Reparent
+
+Similar to [Ownership](../netcode-basics/networkobject#ownership), only the server (or host, which is both a server and a client at the same time) can control reparenting of a `NetworkObject` in the network.
+
+Clients however, can send RPCs to server and execute a logic server-side that ultimately makes server to reparent a `NetworkObject`.
+
+### Only Reparenting Under A `NetworkObject` (Or To The Root) Is Valid
+
+A `NetworkObject` can only be reparented under another `NetworkObject` (`GameObject` with `NetworkObject` component attached). Only exception is moving a `NetworkObject` to the root of the scene hierarchy.
+
+This is simply due to the fact that MLAPI would not be able to identify & locate new parent on the remote-side if it was a non-`NetworkObject` parent. Again, except moving it to the root because we could identify no parent (root) scenario without `NetworkObject` identification or scene hierarchy traversal.
+
+### Only Reparenting During Networking Is Valid
+
+A `NetworkObject` can only be reparented while networking, in other terms you can only reparent while listening/running as a server.
+
+If you allowed moves while not networking, you would be desynced immediately upon switching to networking. Also reparenting a `NetworkObject` under a non-`NetworkObject` while not networking would sound valid but that would not be replicable on the remote-side since Netcode does not cover full scene hierarchy synchronization.
+
+
+### Invalid Reparenting Will Move `NetworkObject` Back To Its Original Location
+
+If an invalid/unsupported `NetworkObject` parenting happens, Netcode will immediately pop it back to its previous location to keep things in sync and also will provide relevant error/warning messages to indicate the issue.
+
+## (Re)Parenting Move Examples
+
+We will assume that our initial scene hierarchy is looking like this:
+
+```
+Sun
+Tree
+Camera
+Player (NetworkObject)
+  ├─ Head
+  ├─ Body
+  ├─ Arms
+  │  ├─ LeftArm
+  │  │  └─ LeftHand (NetworkObject)
+  │  └─ RightArm
+  │     └─ RightHand (NetworkObject)
+  └─ Legs
+Axe (NetworkObject)
+```
```

Comment:
Last time I tried implementing something similar, I noticed child NetworkObjects' OnNetworkSpawn() wouldn't be invoked, which could be a necessity for users.

Unless that has since changed, I would probably include this note. @MFatihMAR, thoughts?

Reply:
I'm not sure why we'd mention this here. The example talks about the ability of reparenting NetworkObjects underneath other NetworkObjects, in this case, Axe will be moved under the RightHand a few steps below. We don't necessarily need to make that action on NetworkBehaviour.OnNetworkSpawn() callback — it could be any other time arbitrarily.